content/modules/r/pages/merge.adoc

```diff
@@ -0,0 +1,302 @@
+----
+myDF <- read.csv("/class/datamine/data/flights/subset/1995.csv")
+----
+
+Here is a listing of the information about the airports themselves:
+
+[source, R]
+----
+airportsDF <- read.csv("/class/datamine/data/flights/subset/airports.csv")
+----
+
+We see that the 3-letter codes about the airports are given in the Origin and Dest columns of `myDF`.
+
+[source, R]
+----
+head(myDF)
+----
+
+It is harder to tell which column in the `airportsDF` gives the 3-letter codes, but these are the iata codes.
+
+[source, R]
+----
+head(airportsDF)
+----
+
+It is perhaps easier to see this from the tail of airportsDF:
+
+[source, R]
+----
+tail(airportsDF)
+----
+
+Now we merge the two data frames, and we display the information about the `Origin` airport, by linking the `Origin` column of `myDF` with the `iata` column of `airportsDF`:
+
+[source, R]
+----
+mynewDF <- merge(myDF, airportsDF, by.x="Origin", by.y="iata")
+----
+
+The resulting data frame has the same size as `myDF`:
+
+[source, R]
+----
+dim(myDF)
+dim(mynewDF)
+----
+
+but now has extra columns, namely, with information about the `Origin` airport:
+
+[source, R]
+----
+head(mynewDF)
+tail(mynewDF)
+----
+
+So now we can do things like calculating a `sum` of all `Distances` of flights with `Origin` in each `state`:
+
+[source, R]
+----
+sort(tapply( mynewDF$Distance, mynewDF$state, sum ))
+----
+
+== Here is another `merge` example:
+
+=== Examples
+
+Consider the data.frame's `books` and `authors`:
+
+[source, R]
+----
+books
+----
+
+[source, R]
+----
+   id                                     title author_id rating
+1   1     Harry Potter and the Sorcerers Stone          1   4.47
+2   2   Harry Potter and the Chamber of Secrets         1   4.43
+3   3  Harry Potter and the Prisoner of Azkaban         1   4.57
+4   4       Harry Potter and the Goblet of Fire         1   4.56
+5   5 Harry Potter and the Order of the Phoenix         1   4.50
+6   6    Harry Potter and the Half Blood Prince         1   4.57
+7   7      Harry Potter and the Deathly Hallows         1   4.62
+8   8                          The Way of Kings         2   4.64
+9   9                            The Book Thief         3   4.37
+10 10                      The Eye of the World         4   4.18
+----
+
+[source, R]
+----
+authors
+----
+
+[source, R]
+----
+   id                     name avg_rating
+1   1             J.K. Rowling       4.46
+2   2        Brandon Sanderson       4.39
+3   3             Markus Zusak       4.34
+4   4            Robert Jordan       4.18
+5   5          Agatha Christie       4.00
+6   6                Alex Kava       4.02
+7   7    Nassim Nicholas Taleb       3.99
+8   8              Neil Gaiman       4.13
+9   9            Stieg Larsson       4.16
+10 10 Antoine de Saint-Exupéry       4.30
+----
+
+==== How do I merge the author information from `authors` based on `author_id` in `books` and `id` in `authors`, keeping only information from authors and books where there is a match?
```

Comment:
How do I merge both data.frames based on the `author_id` column in the `books` data and the `id` column in the `authors` data, keeping only rows where there's a match?